buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -226,46 +286,79 @@ synchronized void start() {
+        try (InputStream byteArrayInputStream = new ByteArrayInputStream(input.getBytes(StandardCharsets.UTF_8))) {
+            services.loggedExec(spec -> {
+                spec.setEnvironment(getESEnvironment());
+                spec.workingDir(workingDir);
+                spec.executable(
+                    OS.conditionalString()
+                        .onUnix(() -> "./bin/" + tool)
+                        .onWindows(() -> "cmd")
+                        .supply()
+                );
+                spec.args(
+                    OS.<List<String>>conditional()
+                        .onWindows(() -> {
+                            ArrayList<String> result = new ArrayList<>();
+                            result.add("/c");
+                            result.add("bin\\" + tool + ".bat");
+                            for (String arg : args) {
+                                result.add(arg);
+                            }
+                            return result;
+                        })
+                        .onUnix(() -> Arrays.asList(args))
+                        .supply()
+                );
+                spec.setStandardInput(byteArrayInputStream);
+
+            });
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
     private void runElaticsearchBinScript(String tool, String... args) {
-        services.loggedExec(spec -> {
-            spec.setEnvironment(getESEnvironment());
-            spec.workingDir(workingDir);
-            spec.executable(
-                OS.conditionalString()
-                    .onUnix(() -> "./bin/" + tool)
-                    .onWindows(() -> "cmd")
-                    .supply()
-            );
-            spec.args(
-                OS.<List<String>>conditional()
-                    .onWindows(() -> {
-                        ArrayList<String> result = new ArrayList<>();
-                        result.add("/c");
-                        result.add("bin\\" + tool + ".bat");
-                        for (String arg : args) {
-                            result.add(arg);
-                        }
-                        return result;
-                    })
-                    .onUnix(() -> Arrays.asList(args))
-                    .supply()
-            );
-        });
+        runElaticsearchBinScriptWithInput("", tool, args);
     }
 
     private Map<String, String> getESEnvironment() {
-        Map<String, String> environment= new HashMap<>();
-        environment.put("JAVA_HOME", getJavaHome().getAbsolutePath());
-        environment.put("ES_PATH_CONF", configFile.getParent().toString());
-        environment.put("ES_JAVA_OPTS", "-Xms512m -Xmx512m");
-        environment.put("ES_TMPDIR", tmpDir.toString());
+        Map<String, String> defaultEnv = new HashMap<>();
+        defaultEnv.put("JAVA_HOME", getJavaHome().getAbsolutePath());
```

Comment:
This means we are never testing with the bundled jdk. I think we should have the same logic ClusterFormationTasks does to decide whether to add JAVA_HOME to the env.

Reply:
I was planning to address that as part of #40531